stacks-common/src/util/log.rs

```diff
@@ -215,35 +215,23 @@ fn make_json_logger() -> Logger {
     panic!("Tried to construct JSON logger, but stacks-blockchain built without slog_json feature enabled.")
 }
 
-#[cfg(not(any(test, feature = "testing")))]
 fn make_logger() -> Logger {
     if env::var("STACKS_LOG_JSON") == Ok("1".into()) {
         make_json_logger()
     } else {
         let debug = env::var("STACKS_LOG_DEBUG") == Ok("1".into());
         let pretty_print = env::var("STACKS_LOG_PP") == Ok("1".into());
+        #[cfg(not(any(test, feature = "testing")))]
         let decorator = slog_term::PlainSyncDecorator::new(std::io::stderr());
+        #[cfg(any(test, feature = "testing"))]
+        let decorator = slog_term::PlainSyncDecorator::new(slog_term::TestStdoutWriter);
```

Comment:
I think the canonical way to do this is a little cleaner:

```rust
let decorator = {
    #[cfg(not(any(test, feature = "testing")))]
    {
        slog_term::PlainSyncDecorator::new(std::io::stderr())
    }
    #[cfg(any(test, feature = "testing"))]
    {
        slog_term::PlainSyncDecorator::new(slog_term::TestStdoutWriter)
    }
};
```

Reply:
Even better would be to put the entire declaration block into its own function, and have one function for testing and one for prod.

Reply:
Good call, I pulled it out into its own function 👍🏼